### PR TITLE
feat(useElementVisibility): support watch the real element

### PR DESCRIPTION
**Pull request by sxzz**
relate #2160

### Description

Cannot get the real HTMLElement, we should use `unrefElement`.

Also, if `element` is changed, it should also be updated.

### Additional context

---

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [ ] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [ ] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Review by songjianet — DISMISSED** [review]
LGTM

File: packages/core/useElementVisibility/index.ts [hunk 1]
```diff
@@ -43,12 +44,19 @@ export function useElementVisibility(
...
 
   if (window) {
-    tryOnMounted(() => useEventListener(
-      () => resolveUnref(scrollTarget) || window, 'scroll', testBounding, { capture: false, passive: true }))
+    useEventListener(scrollTarget || window, 'scroll', testBounding, {
+      capture: false, passive: true,
+    })
   }
 
+  tryOnScopeDispose(() => stop())
```
Comment (antfu) [inline packages/core/useElementVisibility/index.ts:59]: this is redundant, `watch` already did it.

Comment (sxzz) [reply packages/core/useElementVisibility/index.ts]: Oops, just updated.

**Comment by mymx2** [comment]
also remove (el as HTMLElement)
`
const rect = (el as HTMLElement).getBoundingClientRect()
`
to
`
const rect = el.getBoundingClientRect()
`

**Comment by mymx2** [comment]
and what if use watch instead of tryOnMounted. also Considering use v-if both on the Vue component instance and component.
```
const el = resolveUnref(element)
if (!el) {
  elementIsVisible.value = false
}
```
el instanceof Element
```
const el = unrefElement(element)
if (!el || !(el instanceof Element)) {
  elementIsVisible.value = false
}
```

**Comment by sxzz** [comment]
@mymx2 `MaybeComputedElementRef` it must be a Element, so I think we needn't have repeat judgement in runtime code.

**Review by antfu — APPROVED** [review]

**Pull request merged**